gems/sorbet-runtime/lib/types/private/methods/call_validation.rb

```diff
@@ -161,6 +161,7 @@ def self.disable_fast_path
   end
 
   def self.create_validator_method(mod, original_method, method_sig, original_visibility)
+    T::Private::DeclState.current.is_re_wrapping = true
```

Comment:
This allows us to know that the call to `_on_method_added` that's about to happen is merely a result of the fact that we do some of the sig-wrapping logic lazily, and we therefore shouldn't do the final checks right now. The way we communicate this fact, however, is via global mutable state, which is generally not good.

Reply:
Resolved offline: Not the greatest, but it'll do.